### PR TITLE
Removing PathSemiMasker unnecessary mask node

### PR DESCRIPTION
**Pull request by ted-wq-x**
# Description

As the title suggest.

# Contributor agreement

- [x] I have read and agree to the [Contributor Agreement](https://github.com/kuzudb/kuzu/blob/master/CLA.md).

### CODE REVIEW

File: src/include/planner/operator/sip/logical_semi_masker.h [hunk 1]
```diff
@@ -79,6 +85,8 @@ class LogicalSemiMasker : public LogicalOperator {
     std::vector<common::table_id_t> nodeTableIDs;
     // Operators accepting semi masker
     std::vector<LogicalOperator*> targetOps;
+    // for path semi masker
```
Comment (andyfengHKU) [inline src/include/planner/operator/sip/logical_semi_masker.h:88]: `For path semi masker marking src or dst nodes`

File: src/include/planner/operator/sip/logical_semi_masker.h [hunk 2]
```diff
@@ -62,14 +64,18 @@ class LogicalSemiMasker : public LogicalOperator {
     void addTarget(LogicalOperator* op) { targetOps.push_back(op); }
     std::vector<LogicalOperator*> getTargetOperators() const { return targetOps; }
 
+    void setDirection(common::ExtendDirection extendDirection) { direction = extendDirection; }
```
Comment (andyfengHKU) [inline src/include/planner/operator/sip/logical_semi_masker.h:67]: Better to also assert keyType is `PATH` when setting direction

**Review by andyfengHKU — APPROVED** [review]
Make sense to me. Thanks!

**Comment by ted-wq-x** [comment]
Done

**Pull request merged**